### ISSUE TITLE
Error: context deadline exceeded context deadline exceeded Error: plugin "diff" exited with error

### ISSUE THREAD

**Issue by zzarkouna** [databus23/helm-diff#167]
helm version 3.0.0
helm-diff plugin 3.0.0-rc.7

helm diff upgrade myAPP \
                        --namespace=mynamespace  \
                        --set global.image.tag="myimagetag" \
                        --set kong.front-gw.plugins.jwt.anonymousId="xxxx" \
                        --values=values.yaml \
                        --values=values-u.yaml \
                        ../helm/myAPP

i get this error:
Error: context deadline exceeded
context deadline exceeded
Error: plugin "diff" exited with error

**Comment by databus23** [comment]
Can you confirm that `helm get manifest myAPP` on the other hand works?

Also could you try to find would actually times out by running the diff command with `--debug`

**Comment by zzarkouna** [comment]
helm get manifest myAPP --debug
Error: release: not found
helm.go:76: [debug] release: not found
helm.sh/helm/v3/pkg/storage/driver.init
	/home/circleci/helm.sh/helm/pkg/storage/driver/driver.go:27
runtime.doInit
	/usr/local/go/src/runtime/proc.go:5222
runtime.doInit
	/usr/local/go/src/runtime/proc.go:5217
runtime.doInit
	/usr/local/go/src/runtime/proc.go:5217
runtime.doInit
	/usr/local/go/src/runtime/proc.go:5217
runtime.main
	/usr/local/go/src/runtime/proc.go:190
runtime.goexit
	/usr/local/go/src/runtime/asm_amd64.s:1357

**Comment by databus23** [comment]
ok this looks like helm can talk to your cluster. Can you try running the actual `helm diff upgrade` command with `--debug`? Hopefully we can then find out where its timing out

**Comment by zzarkouna** [comment]
Error: context deadline exceeded
context deadline exceeded
Error: plugin "diff" exited with error
helm.go:76: [debug] plugin "diff" exited with error

**Comment by databus23** [comment]
This is odd, there should be more output. Is this everything the diff command printed when run with
  `--debug`

**Comment by zzarkouna** [comment]
this output was with --debug flag there is no other output!

**Comment by databus23** [comment]
Can you double check the output of
`helm version`
`helm plugin list`

If you are using helm 3 and version 3.0.0-rc.7 there should be some more output.

**Comment by zzarkouna** [comment]
```
helm version
version.BuildInfo{Version:"v3.0.0", GitCommit:"e29ce2a54e96cd02ccfce88bee4f58bb6e2a28b6", GitTreeState:"clean", GoVersion:"go1.13.4"}
```

```
helmplugin list
NAME	VERSION   	DESCRIPTION
2to3	0.2.0     	migrate and cleanup Helm v2 configuration and releases in-place to Helm v3
diff	3.0.0-rc.7	Preview helm upgrade changes as a diff
gcs 	0.2.2     	Manage repositories on Google Cloud Storage
```

```
helm diff upgrade myAPP --debug \
                        --namespace=mynamespace  \
                        --set global.image.tag="myimagetag" \
                        --values=values.yaml \
                        --values=values-u.yaml \
                        ../helm/myAPP
Error: context deadline exceeded
context deadline exceeded
Error: plugin "diff" exited with error
helm.go:76: [debug] plugin "diff" exited with error
```

**Comment by thangbn** [comment]
i repeatly get this error on latest helm and helm-diff plugin version

**Comment by databus23** [comment]
Strange. I can't reproduce and the debug output is not helping so far.

**Comment by psucodervn** [comment]
I'm using 3.0.0-rc.7 and facing the same problem. Rollback to 3.0.0-rc.6 get it works for me. Just don't know why...

**Comment by databus23** [comment]
@psucodervn that is super weird, there are no changes between these two versions that could explain this regression.

**Comment by psucodervn** [comment]
> How long das the command take until this error appears? Can somebody who can reproduce this run the helm command with `time` in front?

I've recent update to 3.0.0-rc.7 and the error re-appear 😂
FYI:
```
$ time helm diff rule-editor charts/rule-editor
Command "helm diff" is deprecated, use "helm diff upgrade" instead
Error: context deadline exceeded
context deadline exceeded
Error: plugin "diff" exited with error
helm diff rule-editor charts/rule-editor  0.05s user 0.02s system 0% cpu 30.056 total
```

**Comment by databus23** [comment]
Ok I have a theory. Could you also run `helm diff version` and check that the reported version by the binary is matching the plugin in version.
This may be related to #166.
In general the 30 second timeout is suspicious because the only timeout I can find in the code is in the codepath for helm2: https://github.com/databus23/helm-diff/blob/977b5c822f660a37c127e0a11b4b08694b1de306/cmd/helpers.go#L53

**Comment by psucodervn** [comment]
> Ok I have a theory. Could you also run `helm diff version` and check that the reported version by the binary is matching the version reported by `helm plugin list`.
> This may be related to #166.


Yeah, it's seems to be the same problem with #166
```
$ helm plugin install https://github.com/databus23/helm-diff
$ helm plugin list
NAME	VERSION   	DESCRIPTION
diff	3.0.0-rc.7	Preview helm upgrade changes as a diff
$ helm diff version
2.11.0+5
```
If I explicitly install the version 3.0.0-rc.7, it works well, and the `helm diff` command has been succeed 
```
$ helm plugin install https://github.com/databus23/helm-diff --version 3.0.0-rc.7
$ helm plugin list
NAME	VERSION   	DESCRIPTION
diff	3.0.0-rc.7	Preview helm upgrade changes as a diff
$ helm diff version
3.0.0-rc.7
$ time helm diff rule-editor charts/rule-editor
Command "helm diff" is deprecated, use "helm diff upgrade" instead
helm diff rule-editor charts/rule-editor  1.78s user 0.35s system 42% cpu 4.993 total
```

**Comment by databus23** [comment]
@psucodervn Awesome, thanks for finding this out. The installation script always has been a pain.

So to everyone affected by this please specify the plugin version when installing. Don't relyon `latest`.

`helm plugin install https://github.com/databus23/helm-diff --version 3.0.0-rc.7`

**Comment by AlexShemeshWix** [comment]
Ok ill check more.
I have 2 versions of helm. 
1. helm v2.16.1
2. helm3 v3.0.0
I have huge infrastructure project that i need to migrate from helm to helm3.
Installing diff with specific version for helm3 only didnt helped. 
I had to remove diff plugin from both helm and helm3 and install same version of diff plugin
```
helm3 plugin install https://github.com/databus23/helm-diff --version v3.0.0-rc.7
helm plugin install https://github.com/databus23/helm-diff --version v3.0.0-rc.7
```
After that helm3 diff worked as expected.

**Comment by AlexShemeshWix** [comment]
Is it fixed?

**Comment by deiga** [comment]
> Is it fixed?

https://github.com/databus23/helm-diff/issues/167#issuecomment-558535405

**Comment by databus23** [comment]
This should be fixed since 3.1.0 has been released
